express/index.cjs

```diff
@@ -140,25 +161,174 @@ app.post('/register/employer', async (req, res) => {
+  writer.write(`${setTimestamp(newTime)} | login attempt: seeker\n`);
+  const {cred, pass} = req.body;
+  try {
+    let check;
+    if(cred == null || pass == null) {
+      throw({status: 400, error: 'failed seeker login', reason: 'missing field'});
+    }
+    if(
+      !validSAN(cred) ||
+      !validSAN(pass)
+    ) {
+      throw({status: 400, error: 'failed seeker login', reason: 'invalid input'});
+    }
+    check = await checkUser(req, cred, cred, `Seeker`);
+    if(check.exists == false) {
+      throw({status: 400, error: 'failed seeker login', reason: 'user not found'});
+    }
+    let response;
+    if(!cred.includes('@')) {
+      response = await req.db.query(`SELECT user_name, user_pass, email, hex(seeker_id) AS user_id FROM Seeker
+        WHERE (user_name = :cred AND delete_flag = 0);`,
+        {cred: cred}
+      );
+    } else {
+      response = await req.db.query(`SELECT user_name, user_pass, email, hex(seeker_id) AS user_id FROM Seeker
+        WHERE email = :cred AND delete_flag = 0;`,
+        {cred: cred}
+      );
+    }
+    const [[users]] = response;
+    if (!users) {
+      throw({status: 500, error: 'failed login', reason: 'user does not exist'});
+    }
+    const dbPassword = `${users.user_pass}`;
+    const compare = await bcrypt.compare(pass, dbPassword);
+    if(!compare) {
+      throw({status: 400, error: 'failed login',reason: 'incorrect password'});
+    }
+    const payload = {
+      user_id: users.user_id,
+      user_name: users.user_name,
+      email: users.email,
+      exp: timeNow + (60 * 60 * 24 * 7 * 2)
+    }
+    
+    const encodedUser = jwt.sign(payload, process.env.JWT_KEY);
+    
+    res.status(200).json({success: true, user_name: users.user_name, email: users.email, jwt: encodedUser});
+    writer.write(`${setTimestamp(newTime)} | status: 200 | source: /login/seeker | login: seeker ${users.user_name} logged in | @${req.socket.remoteAddress}\n`);
+  } catch (err) {
+    console.warn(err);
+    if(!err.reason) {
+      res.status(500).json({success: false, error: 'server failure'});
+      writer.write(`${setTimestamp(newTime)} | status: 500 | source: /login/seeker | error: ${err} | attempt: ${username}@${req.socket.remoteAddress}\n`);
+    }
+    else {
+      res.status(!err.status ? 500 : err.status).json({success: false, error: err.reason});
+      writer.write(`${setTimestamp(newTime)} | status: ${!err.status ? 500 : err.status} | source: /login/seeker | error: ${err.error} | reason: ${err.reason} | attempt: ${username}@${req.socket.remoteAddress}\n`);
+    }
+  }
+});
+
+app.post('/login/employer', async (req, res) => {
+  console.log('login attempt: employer');
+  const timeNow = Math.ceil(Date.now() / 1000);
+  const newTime = new Date(Date.now());
+  writer.write(`${setTimestamp(newTime)} | login attempt: employer\n`);
+  const {cred, pass, company} = req.body;
+  try {
+    let check;
+    if(cred == null || pass == null || company == null) {
```

Comment:
We should almost always use `===` instead of `==` when checking for equality. You might also be able to jsut use the `!` operate here. For example `if(!cred || !pass || !company) {`